packages/assistant/src/RPA/Assistant/library.py

```diff
@@ -0,0 +1,991 @@
+            ...    rows=5
+            ${result}=    Run dialog
+            Send feedback message    ${result.email}  ${result.message}
+
+        Dialog as progress indicator
+            Add heading    Please wait while I open a browser
+            ${dialog}=     Show dialog    title=Please wait    on_top=${TRUE}
+            Open available browser    https://robocorp.com
+            Close dialog   ${dialog}
+    """
+
+    def __init__(self) -> None:
+        self.logger = logging.getLogger(__name__)
+        self._client = FletClient()
+
+        try:
+            # Prevent logging from keywords that return results
+            keywords = [
+                "Run dialog",
+            ]
+            BuiltIn().import_library(
+                "RPA.core.logger.RobotLogListener", "WITH NAME", "RPA.RobotLogListener"
+            )
+            listener = BuiltIn().get_library_instance("RPA.RobotLogListener")
+            # useful to comment out when debugging
+            listener.register_protected_keywords(keywords)
+        except RobotNotRunningError:
+            pass
+
+    def _add_closing_button(self, label="Submit") -> None:
+        def close(e):
+            self._client.page.window_destroy()
+
+        self._client.add_element(ElevatedButton(label, on_click=close))
+
+    @keyword("Clear elements")
+    def clear_elements(self) -> None:
+        """Remove all previously defined elements and start from a clean state
+
+        By default this is done automatically when a dialog is created.
+
+        Example:
+
+        .. code-block:: robotframework
+
+            Add heading     Please input user information
+            FOR    ${user}   IN    @{users}
+                Run dialog    clear=False
+                Process page
+            END
+            Clear elements
+        """
+        self._client.clear_elements()
+
+    @keyword("Add heading")
+    def add_heading(
+        self,
+        heading: str,
+        size: Size = Size.Medium,
+    ) -> None:
+        """Add a centered heading text element
+
+        :param heading: The text content for the heading
+        :param size:    The size of the heading
+
+        Supported ``size`` values are Small, Medium, and Large. By default uses
+        the value Medium.
+
+        Example:
+
+        .. code-block:: robotframework
+
+            Add heading     User information  size=Large
+            Add heading     Location          size=Small
+            Add text input  address           label=User address
+            Run dialog
+        """
+        if not isinstance(size, Size):
+            size = Size(size)
+
+        if size == Size.Small:
+            self._client.add_element(element=Text(heading, style="headlineSmall"))
+        elif size == Size.Medium:
+            self._client.add_element(element=Text(heading, style="headlineMedium"))
+        elif size == Size.Large:
+            self._client.add_element(element=Text(heading, style="headlineLarge"))
```

Comment:
This (and other similar methods) would maybe benefit from only mapping the style string in these conditionals, and having a single add_element call, just to make reading and refactoring easier. It can also be done with a single dict lookup, if so inclined:
```python
style = {
    Size.Small: "headlineSmall",
    Size.Medium: "headlineMedium",
    Size.Large: "headlineLarge",
}[size]

self._client.add_element(element=Text(heading, style=style))
```

Reply:
yep, totally agreed with this, wanted to refactor it like this but kind of postponed all refactoring until after the release

Reply:
done